Shared/lib/Dancer/Plugin/Database/Core/Handle.pm

```diff
@@ -365,43 +365,13 @@ sub _generate_sql {
 
     if ($type eq 'UPDATE' || $type eq 'DELETE' || $type eq 'SELECT' || $type eq 'COUNT')
     {
-        if ($where && !ref $where) {
-            $sql .= " WHERE " . $where;
-        } elsif ( ref $where eq 'HASH' ) {
-            my @stmts;
-            foreach my $k ( sort keys %$where ) {
-                my $v = $where->{$k};
-                if ( ref $v eq 'HASH' ) {
-                    my $not = delete $v->{'not'};
-                    while (my($op,$value) = each %$v ) {
-                        my ($cond, $add_bind_param) 
-                            = $self->_get_where_sql($op, $not, $value);
-                        push @stmts, $self->_quote_identifier($k) . $cond; 
-                        push @bind_params, $v->{$op} if $add_bind_param;
-                    }
-                } else {
-                    my $clause .= $self->_quote_identifier($k);
-                    if ( ! defined $v ) {
-                        $clause .= ' IS NULL';
-                    }
-                    elsif ( ! ref $v ) {
-                        $clause .= '=?';
-                        push @bind_params, $v;
-                    }
-                    elsif ( ref $v eq 'ARRAY' ) {
-                        $clause .= ' IN (' . (join ',', map { '?' } @$v) . ')';
-                        push @bind_params, @$v;
-                    }
-                    push @stmts, $clause;
-                }
-            }
-            $sql .= " WHERE " . join " AND ", @stmts if keys %$where;
-        } elsif (ref $where) {
-            carp "Can't handle ref " . ref $where . " for where";
-            return;
+        if ($where) {
+            my ($where_sql, @where_binds) = $self->_generate_where_clauses( $where );
+            return unless $where_sql;   # there was a problem
```

Comment:
I wonder if we shouldn't carp here, rather than silently return undef - if we didn't generate any SQL something was clearly very wrong.